spec/features/user_visits_show_page_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+feature "user visits show page" do
+  given(:alice) { Fabricate(:user) }
+  given(:category) { Fabricate(:category) }
+  given(:video) { Fabricate(:video, category_id: category.id) }
+
+  background do
+    sign_in(alice)
+  end
+
+  scenario "when unauthenticated, is redirected to sign in page" do
+    sign_out 
+    visit_show_page(alice)
+    expect(current_path).to eq('/sign_in')
```

Comment:
You should use named routes instead of hard coded paths. that way even if you change the url for route, the spec will pass. You can just compare against `sign_in_path` if you add `as: :sign_in` to the route.